QuizApp/Presentation/Common/UIComponents/DifficultyIndicator.swift

```diff
@@ -0,0 +1,74 @@
+import UIKit
+
+class DifficultyIndicator: UIView {
+
+    struct CustomConstants {
+        static let numberOfRectangles = 3
+        static let defaultSpacing = 4
+        static let rotationAmoutDegrees = 90.0
+        static let totalDegrees = 360.0
+        static let xRotation = 0.0
+        static let yRotation = 0.0
+        static let zRotation = 1.0
+        static let xTranslation = 0.0
+        static let yTranslation = 0.0
+        static let zTranslation = 0.0
+        static let rectDiagonal = 15.0
+        static let indicatorTransparency = 0.3
+        static let cornerRadius = 2
+    }
+
+    private let difficulty: Difficulty
+    private let accentColor: UIColor
+    private let rectContainer: CGRect = CGRect(x: 0, y: 0, width: 10, height: 10)
+
+    init(difficulty: Difficulty = .easy, accentColor: UIColor = .accentYellow) {
+        self.difficulty = difficulty
+        self.accentColor = accentColor
+
+        super.init(frame: .zero)
+
+        drawRectangles()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    private func drawRectangles(spacing: Int = CustomConstants.defaultSpacing) {
+        var translate = CustomConstants.xTranslation
+
+        for index in 1...CustomConstants.numberOfRectangles {
+            let bezierPath = UIBezierPath(
```

Comment:
This is ok but couldn't you just use UIView and rotate it? This seems a bit too much. Also, you could use stack view to order them.
Your choice.

Reply:
Performance wise, this is rendering on GPU, while UIViews render on CPU which is little bit more expensive... and I wanted to have fun with CoreGraphics